std/math.d

```diff
@@ -6798,3 +6799,205 @@ real yl2xp1(real x, real y) @nogc @safe pure nothrow;       // y * log2(x + 1)
     auto x = floor(1.2);
     auto y = ceil(1.2);
 }
+
+/***********************************
+ * Defines a total order on all floating-point numbers.
+ *
+ * The order is defined as follows:
+ * $(UL
+ *      $(LI All numbers in [-$(INFIN), +$(INFIN)] are ordered
+ *          the same way as by built-in comparison, with the exception of
+ *          -0.0, which is less than +0.0;)
+ *      $(LI If the sign bit is set (that is, it's 'negative'), $(NAN) is less
+ *          than any number; if the sign bit is not set (it is 'positive'),
+ *          $(NAN) is greater than any number;)
+ *      $(LI $(NAN)s of the same sign are ordered by the payload ('negative'
+ *          ones - in reverse order).)
+ * )
+ *
+ * Returns:
+ *      -1 if $(D x) precedes $(D y) in the order specified above;
+ *      0 if $(D x) and $(D y) are identical, and 1 otherwise.
+ *
+ * See_Also:
+ *      $(MYREF isIdentical)
+ * Standards: Conforms to IEEE 754-2008
+ */
+int cmp(T)(T x, T y) @nogc @trusted pure nothrow
+    if (isFloatingPoint!T)
+{
+    alias F = floatTraits!T;
+
+    static if (F.realFormat == RealFormat.ieeeSingle
+               || F.realFormat == RealFormat.ieeeDouble)
+    {
+        static if (T.sizeof == 4)
+            alias UInt = uint;
+        else
+            alias UInt = ulong;
+
+        enum msb = ~(UInt.max >>> 1);
+
+        UInt*[2] vars = [cast(UInt*)&x, cast(UInt*)&y];
+
+        foreach (var; vars)
+            if (*var & msb)
+                *var = ~*var;
+            else
+                *var |= msb;
+
+        if (*vars[0] < *vars[1])
+            return -1;
+        else if (*vars[0] > *vars[1])
```

Comment:
I went with pointers because virtually everything else in this module uses
pointers. While certainly possible, using unions here would be inconsistent with
the rest of the module.